dvc/repo/experiments/show.py

```diff
@@ -13,15 +15,45 @@
 
 if TYPE_CHECKING:
     from dvc.repo import Repo
+    from dvc.repo.experiments.queue.base import QueueEntry
+    from dvc.repo.experiments.queue.celery import LocalCeleryQueue
+    from dvc_task.proc.process import ProcessInfo
 
 logger = logging.getLogger(__name__)
 
 
+class ExpStatus(Enum):
+    Success = 0
+    Queued = 1
+    Running = 2
+    Failed = 3
+
+
+def _get_last_error_message(
+    celery_queue: "LocalCeleryQueue", exp_rev: str
+) -> str:
+    queue_entry: Optional[
+        "QueueEntry"
+    ] = celery_queue.match_queue_entry_by_name(
+        {exp_rev}, celery_queue.iter_failed()
+    ).get(
+        exp_rev
+    )
+    if queue_entry:
+        try:
+            proc_info: "ProcessInfo" = celery_queue.proc[queue_entry.stash_rev]
+            with open(proc_info.stdout, encoding="utf-8") as f_read:
+                return f_read.readlines()[-1].strip()
+        except (KeyError, FileNotFoundError):
+            pass
+    return ""
```

Comment:
Not sure this is really a reliable way to handle this case. It only works if the pipeline was run without debugging enabled.

@mattseddon can weigh in on this, but I think it would be better to just use a generic "Experiment run failed." error message, and then add a new field where we can pass the log file path to the vscode extension. And then on the vscode side they could potentially display the entire log to the user.

Actually, passing the log location seems useful in every exp case, not just for failures. This is something that the vscode extension could also get through `dvc queue logs` but I think it might be easier for them to just get the log file path and read it themselves when needed

Reply:
My plan on this is to use "type" field in "error" later in which I can add info of in which part (initialization, env preparing, running, result collection) of the tasks the error comes out.

Reply:
In that case I would still suggest just using a generic error message for now instead of trying to use the last log line, since the last log line is not actually guaranteed to contain anything useful.

In the long run we should probably also add an error message field to the executor info and save it here:
https://github.com/iterative/dvc/blob/d2a31eb6603ad65188cb6451d8a88d1fc8c67b01/dvc/repo/experiments/executor/base.py#L568-L580

So on getting an exception during repro, we would just do something like
```
info.error = str(exc)
```

and then that would be included in the dump:
https://github.com/iterative/dvc/blob/d2a31eb6603ad65188cb6451d8a88d1fc8c67b01/dvc/repo/experiments/executor/base.py#L494

This would give you a more reliable error message for the failed exp case

Reply:
> @mattseddon can weigh in on this, but I think it would be better to just use a generic "Experiment run failed." error message, and then add a new field where we can pass the log file path to the vscode extension. And then on the vscode side they could potentially display the entire log to the user.
>
> Actually, passing the log location seems useful in every exp case, not just for failures. This is something that the vscode extension could also get through dvc queue logs but I think it might be easier for them to just get the log file path and read it themselves when needed

Yes, the log location would be helpful. 

The current approach for any error is to display the provided error as a tooltip in the experiments table. E.g:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/37993418/191179267-583197de-216d-4fc1-9338-da5e6734501a.png">

It is unlikely that we would want to put the entire log into a tooltip so we would have to change the approach there. It would be good to get as much information about the error as possible without having to parse the logs.